### PR TITLE
Allow advanced setup for stylelint core and plugin pack authors

### PR DESCRIPTION
**Pull request by hudochenkov**
I've tested in with stylelint core and with `stylelint-order`. It works great!

With `stylelint-order` I have problems with my tests, because some of them doesn't have `message` for `rejected`, or doesn't check for multiple warnings (we added this not so long ago to stylelint Jest setup).

Removed mock for `getOsEol` because it's something that is needed in stylelint core only.

Closes https://github.com/stylelint/jest-preset-stylelint/issues/16.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,6 +13,7 @@
   "author": "stylelint",
   "files": [
     "getOsEol.js",
```
Comment (jeddy3) [inline package.json:15]: We can remove this.

File: package.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
...
+    "getTestRule.js",
     "jest-preset.js",
     "jest-setup.js"
```
Comment (jeddy3) [inline package.json:18]: Should this file be called `jest.setup.js` to be consistent with how we're suggesting authors configure their Jest?

File: README.md [hunk 3]
```diff
@@ -14,6 +14,8 @@ npm install jest-preset-stylelint jest stylelint --save-dev
 
 ## Setup
 
+### Simple setup
```
Comment (jeddy3) [inline README.md:17]: I think all authors should use the preset as it currently sets one additional property, `testEnvironment: 'node'`, and may set more in the future.

The advanced setup would then be using the preset but overriding the `setupFiles` property.

---

## Setup

Add the preset to your `jest.config.js` or `jest` field in `package.json`:

```json
{
  "preset": "jest-preset-stylelint"
}
```

Optionally, you can avoid specifying `plugins` in every schema by defining your own setup file to configure the `testRule` function. This is useful if you have many tests. There are two steps:

1. Create `jest.setup.js` in the root of your project. Provide [`plugins`](#plugins-arraystring) option to `getTestRule()`:

   ```js
   const stylelint = require("stylelint");
   const getTestRule = require("jest-preset-stylelint/getTestRule");

   global.testRule = getTestRule(stylelint, { plugins: ["./"] });
   ```

2. Add `jest.setup.js` to your `jest.config.js` or `jest` field in `package.json`:

   ```json
   {
     "preset": "jest-preset-stylelint",
     "setupFiles": ["jest.setup.js"]
   }
   ```

---

File: package.json (hunk 1 above)
Comment (hudochenkov) [reply package.json:15]: Nice catch :)

**Comment by hudochenkov** [comment]
@jeddy3 I've made proposed changes.

**Comment by hudochenkov** [comment]
I think we can merge this PR. I locally migrated `stylelint-order` from own copy of `jest.setup.js`. Works good.

**Comment by jeddy3** [comment]
> I locally migrated stylelint-order from own copy of jest.setup.js. Works good.

That's great news.

I can release `2.0.0` later today.

**Pull request merged**